meshnet/monitor.go

```diff
@@ -0,0 +1,99 @@
+package meshnet
+
+import (
+	"log"
+	"sync"
+
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/vishvananda/netlink"
+)
+
+type (
+	PID     uint32
+	SetupFn func() (MonitorChannels, error)
+)
+
+type MonitorChannels struct {
+	EventCh chan netlink.ProcEvent
```

Comment:
Maybe channels here can be limited to receive only?

Reply:
Good point. But I only changed `ErrCh` to read-only because I'm using others to simulate processing in tests